doc/python-type-hinting.md

```diff
@@ -40,18 +40,16 @@ def get_objects_matching(ids: Sequence[str]) -> dict[int, dict]
 
 ## Optional arguments
 
-When `Optional` is used for function arguments, it means that the argument can have the value `None`, not that the argument is optional. 
-
-For arguments that can be a single type or `None`, use `Optional`:
+For arguments that can be a single type or None, use the shorthand `|`. This is recommended instead of using older `Optional` keyword:
 
 ```python
-def test(self, var: Optional[int]) -> None:
+def test(self, var: [None | int]) -> None:
 ```
 
-For arguments that can be one of multiple types or `None`, use `Union` with `None`:
+For arguments that can be one of multiple types or None, use the shorthand `|`. This is recommended instead of using older `Union` keyword:
 
 ```python
-def test(self, var: Union[None, int, str]) -> None:
+def test(self, var: [None | int | str]) -> None:
```

Comment:
I'm of the opinion that `Optional` is clearer when there is only one type.

This:
```py
var: Optional[int]
```

instead of: 

```py
var: int | None
```

---

But, this: 

```py
var: int | str | None
``` 

instead of: 

```py
var: Union[Optional[int], Optional[str]]
```

See what the others think. Happy to go with the consensus.

Reply:
Don't have a preference at first glance although:

It would seem Python's core development team [recommends](https://stackoverflow.com/a/69440627) using `Type | None` over `Optional` for Python 10+.
Thus I suspect the piping symbol will be widely adopted moving forward and `Optional` will become outdated.

Using the piping symbol will make it easier for new joiners to get familiar with the code base.

Reply:
My view is that, I think the repeated `None` makes it less readable. The official doc doesn't mention the deprecation of it or preference. I'm only in favour of using `Optional` if it is a single type. If you have 2 types + None, then the new operator would be best.

To me, this looks isn't very clear:

```py
def test(self, var: int | None = None) -> None:
```

compared to:

```py
def test(self, var: Optional[int] = None) -> None:
```

Again, happy to go with consensus if others think it is readable.

Reply:
I prefer if we only use `|`, I find it more readable, even in `int | None` situation, but up to you guys.

Reply:
Looking at the examples I like the idea of using the `|` :
`int | str | None` 

But when there default types
`Optional[t] = None`
is easier to read to me then:
`t | None = None`

Reply:
Okay, in that case let's go with the single use of optional using `pipe` operator i.e `Type | None = None`